### PR TITLE
fix: fix guide & modal focus lock issues

### PR DESCRIPTION
**Pull request by dahaca**
Fix modal not receiving focus when the guide is opened, move focus lock to the `Modal` component to ensure all modals have it.

### CODE REVIEW

**Comment by marekrjpolak** [comment]
There's opposite problem now. I'm unable to search in guide when a modal is displayed. It works only if `isGuideOnTop`.

File: packages/suite/src/components/suite/modals/index.tsx [hunk 1]
```diff
@@ -46,23 +45,27 @@ import { RawRenderer } from '../Modal/RawRenderer';
...
         onPinCancel: allModalActions.onPinCancel,
         onReceiveConfirmation: allModalActions.onReceiveConfirmation,
         goto: routerActions.goto,
     });
+
     return {
         ...props,
         ...actions,
         isGuideOpen,
+        isGuideOnTop,
```
Comment (marekrjpolak) [inline packages/suite/src/components/suite/modals/index.tsx]: Are these 'guide' values still necessary in `useSharedProps`, given that the general `Modal` component obtains them itself?

File: packages/suite/src/components/suite/modals/index.tsx [hunk 2]
```diff
@@ -262,24 +265,26 @@ const getModal = (props: SharedProps) => {
 
 // Action modal container component
 export const Modals = () => {
-    const props = useSharedProps();
+    const props: SharedProps = useSharedProps();
+
     const modalComponent = getModal(props);
+
     if (!modalComponent) {
```
Comment (marekrjpolak) [inline packages/suite/src/components/suite/modals/index.tsx]: This `null` check isn't relevant anymore, am I right?

**Review by marekrjpolak — CHANGES_REQUESTED** [review]

File: packages/suite/src/components/suite/modals/index.tsx (hunk 1 above)
Comment (dahaca) [reply packages/suite/src/components/suite/modals/index.tsx]: done

File: packages/suite/src/components/suite/modals/index.tsx (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/components/suite/modals/index.tsx]: done

**Comment by dahaca** [comment]
> There's opposite problem now. I'm unable to search in guide when a modal is displayed. It works only if `isGuideOnTop`.

Fixed, now they are in the same focus group.

**Review by marekrjpolak — APPROVED** [review]
It works and I really like the focus group idea. :+1: 
Squash it, please, and we'll merge it.

**Pull request merged**